p2p/src/peer_manager/peerdb/address_data/mod.rs

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::{Duration, Instant};
+
+use utils::const_value::ConstValue;
+
+/// When the node drops the unreachable node address. Used for negative caching.
+pub const PURGE_UNREACHABLE_TIME: Duration = Duration::from_secs(3600);
+
+/// When the server drops the unreachable node address that was once reachable. This should take about a month.
+/// Such a long time is useful if the node itself has prolonged connectivity problems.
+pub const PURGE_REACHABLE_FAIL_COUNT: u32 = 35;
+
+pub enum AddressState {
```

Comment:
Are we copy-pasting this from the DNS server?

Reply:
I'm talking about the whole file, btw.

Reply:
The approach is similar, but not the same. Most importantly, addresses in `PeerDb` do not have `Connecting` and `Disconnecting` states. This is because `PeerManager` already has information about pending connection states, and so duplicating it in `PeerDb` is not very good.